packages/amplify_core/lib/src/types/common/amplify_auth_provider.dart

```diff
@@ -15,11 +15,18 @@
 
 import 'package:amplify_core/amplify_core.dart';
 import 'package:aws_signature_v4/aws_signature_v4.dart';
+import 'package:meta/meta.dart';
 
 /// An identifier to use as a key in an [AmplifyAuthProviderRepository] so that
 /// a retrieved auth provider can be typed more accurately.
-class AmplifyAuthProviderToken<T extends AmplifyAuthProvider> extends Token<T> {
-  const AmplifyAuthProviderToken();
+///
+/// Does not extend [Token] because [Token] is equal to another [Token] of same
+/// type which has unintended consequences.
```

Comment:
I was finding that user could not register multiple auth providers that implement same type as configured in enum token, could not register 2 of type `TokenAmplifyAuthProvider` bc they would be considered equal by equality fn in `Token` and thus get overwritten in the repo map when setting.

Another way to go was to have additional no-op abstract classes in core for function/oidc which extend the token class. I tried that first pass and got annoying bc they are essentially same shape. If anyone has other ideas let me know.